src/common/components/Carousel/index.tsx

```diff
@@ -0,0 +1,107 @@
+import {
+  Children,
+  cloneElement,
+  ComponentProps,
+  isValidElement,
+  ReactNode,
+  useRef,
+} from 'react';
+
+import Group from '~/common/components/Group';
+
+import CarouselButton from './CarouselButton';
+import useDragScroll from './useDragScroll';
+
+interface CarouselProps extends ComponentProps<'div'> {
+  children: ReactNode;
+  itemsToShow?: number;
+  childSize?: number;
+  useButton?: boolean;
+  groupGap?: number;
+}
+
+const Carousel = ({
+  children,
+  itemsToShow = 4,
+  childSize = 100,
+  useButton = false,
+  groupGap = 5,
+  className,
+  ...props
+}: CarouselProps) => {
+  const containerRef = useRef<HTMLDivElement | null>(null);
+  const itemToShowWidth = itemsToShow * (childSize + groupGap) + childSize / 2;
+  const totalCarousels = Children.count(children);
+
+  const avatars = Children.toArray(children).map((child: ReactNode) => {
+    if (isValidElement(child)) {
+      return cloneElement(child as JSX.Element, {
+        style: { width: `${childSize}px` },
+        className: `${
+          child.props.className ? child.props.className + ' ' : ''
+        }snap-start`,
+      });
+    }
+  });
+
+  const {
+    isLeftButtonActive,
+    isRightButtonActive,
+    buttonScrollLeft,
+    buttonScrollRight,
+    handleMouseDown,
+    handleMouseMove,
+    handleMouseUp,
+  } = useDragScroll({
+    containerRef,
+    childSize: childSize + groupGap,
+  });
+
+  return (
+    <div className="flex items-center" {...props}>
+      <div
+        ref={containerRef}
+        className={`order-2 flex snap-x scroll-pl-2 overflow-hidden scroll-smooth px-2 py-2 ${className}`}
```

Comment:
`className`에 겹치는 속성이 있을 수 있으니 `cn`을 쓰는 것이 좋아보입니다!

```suggestion
        className={cn('order-2 flex snap-x scroll-pl-2 overflow-hidden scroll-smooth px-2 py-2', className)}
```

Reply:
제가 너무 많이 불필요한 코드를 작성했었네요.. 지적해주셔서 감사합니다!